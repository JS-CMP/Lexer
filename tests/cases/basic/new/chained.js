var x = "component";

var MyNS = {};
MyNS.Widget = function () {
    this.init = function (x) {
        this.x = x;
        return this;
    };
    this.render = function () {
        return "Rendered " + this.x;
    };
    return this;
};

console.log(new MyNS.Widget().init(x).render()); // NEW(MyNS.Widget).init(x).render();
