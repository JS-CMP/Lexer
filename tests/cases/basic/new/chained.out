#include "types/JsAny.hpp"
#include "global/global.hpp"
#include "types/objects/Types.hpp"
#include "customOperators/CustomOperators.hpp"

int main() {JS::Any x, MyNS;x=JS::Any("component");MyNS=JS::Any(std::make_shared<JS::Object>());MyNS["Widget"]=JS::Any(std::make_shared<JS::Function>([&](const JS::Any &thisArg, const JS::Any &arguments) -> JS::Any {thisArg["init"]=JS::Any(std::make_shared<JS::Function>([&](const JS::Any &thisArg, const JS::Any &arguments) -> JS::Any {JS::Any x = arguments["0"];thisArg["x"]=x;return thisArg;return JS::Any();}));thisArg["render"]=JS::Any(std::make_shared<JS::Function>([&](const JS::Any &thisArg, const JS::Any &arguments) -> JS::Any {return JS::Any("Rendered ")+thisArg["x"];return JS::Any();}));return thisArg;return JS::Any();}));console["log"](NEW(MyNS["Widget"])["init"](x)["render"]());}
